In [20]:
import pandas as pd
df=pd.read_csv('StudentsPerformance.csv')
df.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [3]:
df.shape

(1000, 8)

# 1. Find out how many males and females participated in the test.

In [4]:
df['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

518 Females and 482 Males participated in the test.

# 2.What do you think about the students' parental level of education?

In [5]:
df.describe(include=['object'])

gender race/ethnicity parental level of education     lunch  \
count     1000           1000                        1000      1000   
unique       2              5                           6         2   
top     female        group C                some college  standard   
freq       518            319                         226       645   

       test preparation course  
count                     1000  
unique                       2  
top                       none  
freq                       642

The parental level of education of most students is 'some college'.

# 3. Who scores the most on average for math, reading and writing based on Gender and Test preparation course


In [11]:
dg=df[['gender','math score','reading score','writing score']].groupby(['gender']).agg('mean')
dg

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

From the above table we can see that on average females score high in reading and writing, Whereas males score high in maths

In [12]:
dt=df[['test preparation course','math score','reading score','writing score']].groupby(['test preparation course']).agg('mean')
dt

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

From the above table we can understand that the students who completed test preparation course score high in maths, reading and writing when compared to the other

# 4. What do you think about the scoring variation for math, reading and writing based on Gender and Test preparation course

In [13]:
dg=df[['gender','math score','reading score','writing score']].groupby(['gender']).var()
dg

math score  reading score  writing score
gender                                          
female  256.958593     207.677438     220.369327
male    211.889097     200.211010     202.413924

In Females and Males  variance is more for math score 

In [14]:
dt=df[['test preparation course','math score','reading score','writing score']].groupby(['test preparation course']).var()
dt

math score  reading score  writing score
test preparation course                                          
completed                210.884027     183.265864     175.202612
none                     246.668662     213.419851     226.251739

Here also the students who completed the test preparation course and the ones who did not completed the course shows more variation in math score

# 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.


In [16]:
df['math score'].describe()

count    1000.000000
mean       66.001000
std        15.569567
min         0.000000
25%        57.000000
50%        66.000000
75%        77.000000
max       100.000000
Name: math score, dtype: float64

The top 25% of students are the ones who scored more than 77 for maths